pages/src/navigation/sidebar/sidebar.ts

```diff
@@ -59,11 +60,13 @@ export class ESLDemoSidebar extends ESLToggleable {
   }
 
   protected onShow(params: SidebarActionParams) {
+    localStorage.setItem('sidebar-expanded', JSON.stringify(true));
     this._animation = !params.immediate;
     super.onShow(params);
     this.expandActive(params.initiator === 'init');
   }
   protected onHide(params: SidebarActionParams) {
+    localStorage.setItem('sidebar-expanded', JSON.stringify(false));
```

Comment:
Basically, it's better to move this logic to a separate single method `storeState` for both show and hide action.
And we shouldn't store state right before animation because the animation is a primary action that we should do.